File: 191_numbers_of_1_bits/numbers_of_1_bits.py
class Solution:
    def numbers_of_1_bits(self, n: int) -> int:
        count = 0
        while n:
            n &= n - 1
            count += 1
        return count


if __name__ == "__main__":
    solution = Solution()
    n = 11
    print(solution.numbers_of_1_bits(n))

File: 191_numbers_of_1_bits/numbers_of_1_bits.go
package main

import "fmt"

type Solution struct{}

func (s Solution) numbersOf1Bits(n uint32) int {
	count := 0
	for n != 0 {
		n &= n - 1
		count++
	}
	return count
}

func main() {
	s := Solution{}
	fmt.Println(s.numbersOf1Bits(11))
}
